### ISSUE TITLE
Rolling Changeling when set to not preferred

### ISSUE THREAD

**Issue by Deantwo** [tgstation/tgstation#7630]
I was playing on Badger today, and I rolled Changeling two times in a row!
And the best part is that I had not preferred to even be considered for Changeling.
(First times being Changeling)

The server at the second round only had two players ready.

**Other:**
I don't know if it has anything to do with it being my b-day today.

**Suggestion:**
When deciding on what gamemode to start, it should check if any of the readied players actually wanna play as the related antags. If it doesn't already.

**Comment by GunHog** [comment]
You mentioned Badger...was the pop very low? I believe it is possible to be forced into being a changeling if no other readied players have the preference enabled. With how unpopular Changeling is and the typical low population, I can see that as a possibility.

**Comment by RemieRichards** [comment]
A lot of times in low pop the game will pick a gamemode, and will refuse to cancel it and will instead force the antagonist roll on someone who doesn't have it enabled (I've gotten this a few times on Badger with Traitor instead of Ling)

also happy birthday @Deantwo

**Comment by Deantwo** [comment]
Yeah the first of the two rounds I asked the GM to give it to someone self and we all had a great round.
I don't know how many we were at the start of that round though, but it was most likely around 5.

The second round there was only me and one other as I stated.
I decided it was an as good time as any to experiment a little with the Changeling.
Plus there was no admin around for that round so whatever.

But yeah I was wondering if SS13 can detect someone's birthday, as I think BYOND has a birthday field on the profiles. Then again, knowing our players, this would most likely be exploited if something did change.

PS: Thanks @RemieRichards.
